src/aspire/classification/averager2d.py

```diff
@@ -0,0 +1,1018 @@
+            basis_coefficients = self.composite_basis.evaluate_t(
+                self.src.images(0, np.inf)
+            )
+
+        # We want to maintain the original coefs for the base images,
+        #  because we will mutate them with shifts in the loop.
+        original_coef = basis_coefficients[classes[:, 0], :]
+        assert original_coef.shape == (n_classes, self.alignment_basis.count)
+
+        # Loop over shift search space, updating best result
+        for x, y in product(x_shifts, y_shifts):
+            shift = np.array([x, y], dtype=int)
+            logger.debug(f"Computing Rotational alignment after shift ({x},{y}).")
+
+            # Shift the coef representing the first (base) entry in each class
+            #   by the negation of the shift
+            # Shifting one image is more efficient than shifting every neighbor
+            basis_coefficients[classes[:, 0], :] = self.alignment_basis.shift(
+                original_coef, -shift
+            )
+
+            _rotations, _, _correlations = self._bfr_align(
+                classes, reflections, basis_coefficients
+            )
+
+            # Each class-neighbor pair may have a best shift-rot from a different shift.
+            # Test and update
+            improved_indices = _correlations > correlations
+            rotations[improved_indices] = _rotations[improved_indices]
+            correlations[improved_indices] = _correlations[improved_indices]
+            shifts[improved_indices] = shift
+
+            # Restore unshifted base coefs
+            basis_coefficients[classes[:, 0], :] = original_coef
+
+            if (x, y) == (0, 0):
+                logger.debug("Initial rotational alignment complete (shift (0,0))")
+                assert np.sum(improved_indices) == np.size(
+                    classes
+                ), f"{np.sum(improved_indices)} =?= {np.size(classes)}"
+            else:
+                logger.debug(
+                    f"Shift ({x},{y}) complete. Improved {np.sum(improved_indices)} alignments."
+                )
+
+        return rotations, shifts, correlations
+
+
+class ReddyChatterjiAverager2D(AligningAverager2D):
+    """
+    Attempts rotational estimation using Reddy Chatterji log polar Fourier cross correlation.
+    Then attempts shift (translational) estimation using cross correlation.
+
+    When averaging, performs rotations then shifts.
+
+    Note, it may be possible to iterate this algorithm...
+
+    Adopted from Reddy Chatterji  (1996)
+    An FFT-Based Technique for Translation,
+    Rotation, and Scale-Invariant Image Registration
+    IEEE TRANSACTIONS ON IMAGE PROCESSING, VOL. 5, NO. 8, AUGUST 1996
+
+    This method intentionally does not use any of ASPIRE's basis
+    so that it may be used as a reference for more ASPIRE approaches.
+    """
+
+    def __init__(
+        self,
+        composite_basis,
+        source,
```

Comment:
Stick with standard `src`?

Reply:
I sort of like the full real names as args and method names (code can be shorter), but I'll change it to be more consistent with the code base. Probably right thing to do. NP.

Reply:
In general I agree, but this is one that's everywhere and there's already an established convention.

Reply:
(changed)